### PR TITLE
[Tests] Added a test for progress callback when doing upload task

### PR DESCRIPTION
**Pull request by phimage**
a simple test for upload using httpbin.org website

see #1115

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 - Added `AuthorizationType` to `AccessTokenAuthorizable` representing request headers of `.none`, `.basic`, and `.bearer`. 
 - Added tests for `Single<Response>` operators.
 - Added `Progress` object into the response when calling progress callback on completion.
+- Added a test for progress callback when doing upload task.
```
Comment (sunshinejr) [inline Changelog.md]: Yeah, I think we can skip changelog entry here.

File: Tests/MoyaProviderSpec.swift [hunk 2]
```diff
@@ -771,6 +771,48 @@ class MoyaProviderSpec: QuickSpec {
...
 
+        describe("a provider with upload progress tracking") {
+            var provider: MoyaProvider<HTTPBin>!
+            beforeEach {
+                provider = MoyaProvider<HTTPBin>()
+            }
+
+            it("tracks progress of request") {
+
+                let url = Bundle(for: MoyaProviderSpec.self).urlForImageResource("testImage.png")!
```
Comment (sunshinejr) [inline Tests/MoyaProviderSpec.swift]: Hmm, CI is failing on this one `Bundle.urlForImageResource("testImage.png")!`, could you check if it compiles for you locally?

Comment (SD10) [reply Tests/MoyaProviderSpec.swift]: I believe this is failing because I don't think`urlForImageResource` is available on iOS.

Comment (phimage) [reply Tests/MoyaProviderSpec.swift]: yes I test with first target only, mac os, my bad
fixed

File: Changelog.md (hunk 1 above)
Comment (phimage) [reply Changelog.md]: Line removed
I just let [atom](https://atom.io/) remove the useless space in changelog (two lines)

**Comment by sunshinejr** [comment]
Thanks for the fixes, @phimage! Would you be up for cleaning the swiftlint warnings as well? Would be awesome.

**Comment by phimage** [comment]
Any reason to not add swiftlint in project build phrase?

**Comment by sunshinejr** [comment]
@phimage I think we just didn't think about it, but one thing to keep in mind is that Carthage needs special settings for that phase so that lint does work only for our development phase and not in users projects. For reference, you can look at [Quick#693](https://github.com/Quick/Quick/issues/693), [Quick#694](https://github.com/Quick/Quick/pull/694) and [Nimble#409](https://github.com/Quick/Nimble/pull/409)

If you want to take a stab at it, go ahead :)

**Pull request merged**